lib/pbench/server/api/resources/query_apis/datasets/table_of_content.py

```diff
@@ -0,0 +1,233 @@
+            dataset.md5,
+            self.parent,
+        )
+
+        # Retrieve the ES indices that belong to this run_id from the metadata
+        # table
+
+        # indices = self.get_index(dataset, "run-toc")
+        indices = "riya-pbench.v6.run-toc.2021-05"
+        if not indices:
+            self.logger.debug(
+                f"Found no indices matching the prefix run-toc"
+                f"for a dataset {dataset!r}"
+            )
+            abort(HTTPStatus.NOT_FOUND, message="Found no matching indices")
+
+        return {
+            "path": f"/{indices}/_search",
+            "kwargs": {
+                "json": {
+                    "size": self.MAX_SIZE,
+                    "query": {
+                        "bool": {
+                            "must": [],
+                            "filter": [
+                                {
+                                    "bool": {
+                                        "should": [
+                                            {
+                                                "bool": {
+                                                    "should": [
+                                                        {
+                                                            "match_phrase": {
+                                                                "directory": self.parent
+                                                            }
+                                                        }
+                                                    ],
+                                                    "minimum_should_match": 1,
+                                                }
+                                            },
+                                            {
+                                                "bool": {
+                                                    "should": [
+                                                        {
+                                                            "match_phrase": {
+                                                                "parent": self.parent
+                                                            }
+                                                        }
+                                                    ],
+                                                    "minimum_should_match": 1,
+                                                }
+                                            },
+                                        ],
+                                        "minimum_should_match": 1,
+                                    }
+                                },
+                                {"match_phrase": {"run_data_parent": run_id}},
+                            ],
+                            "should": [],
+                            "must_not": {
+                                "regexp": {"directory": f"{self.parent}/[^/]+/[^/]+"}
```

Comment:
Ah!  This exclusion will only match subdirectories of subdirectories, but not subdirectories of those (which is how the unit test that I whined about is working!).

This expression should be:
```
                                "regexp": {"directory": f"{self.parent}/[^/]+/.+"}
```

Reply:
@riya-17, just to be clear, this was my fault, not yours -- my original suggestion for the exclusion filter was "right", but later I suggested the one that you used, which is "insufficient".

Reply:
:)

Reply:
Thank you Webb

Reply:
I think this still needs to be addressed.